thorlcr/thorutil/thormisc.hpp

```diff
@@ -306,22 +306,62 @@ class graph_decl CTimeoutTrigger : public CInterface, implements IThreaded
     virtual bool action() = 0;
 };
 
+// Tracks the current and peak storage used for some files
+class CFileSizeTracker: public CInterface
+{
+    RelaxedAtomic<unsigned __int64> activeSize{0};
+    RelaxedAtomic<unsigned __int64> peakSize{0};
+public:
+    void growSize(unsigned __int64 size)
+    {
+        if (size)
+        {
+            unsigned __int64 newActiveSize = activeSize.add_fetch(size);
+            peakSize.store_max(newActiveSize);
+        }
+    }
+    void shrinkSize(unsigned __int64 size)
+    {
+        if (size)
+            activeSize.fetch_sub(size);
+    }
+    unsigned __int64 queryActiveSize() const
+    {
+        return activeSize.load();
+    }
+    unsigned __int64 queryPeakSize() const
+    {
+        return peakSize.load();
+    }
+};
+
 // simple class which takes ownership of the underlying file and deletes it on destruction
 class graph_decl CFileOwner : public CSimpleInterface, implements IInterface
 {
     OwnedIFile iFile;
+    Linked<CFileSizeTracker> fileSizeTracker;
+    __int64 fileSize = 0;
```

Comment:
trivial: this and noteSpill/growSize/shrinkSize should be dealing in offset_t's really (offset_t is the underlying type of file size, __int64 is the underlying type of stats)

Reply:
minor: noteSpill is now an offset_t, but really everything in CFileSizeTracker should be using offset_t too.